### DISCUSSION TITLE
Pyright Error in Pandas DataFrame Parsing

### DISCUSSION THREAD

**Discussion by GhostlyCipher** [microsoft/pyright#7637] [Q&A] [answered]
When attempting to iterate over rows in a Pandas DataFrame using the iterrows() method and access a specific column value, such as converting it to an integer, Pyright raises a type error. The error message suggests that the argument passed to int() is of an incompatible type, leading to confusion in type inference.

```
import pandas as pd

def parser(data: pd.DataFrame) -> List[List]:
    for _, row in data.iterrows():
        some_id = int(row["row_id"])
```
```
(variable) row: pd.Series
```

Error
```
error: Argument of type "Series | Any | ExtensionArray | Unknown | ndarray[Unknown, Unknown]" cannot be assigned to parameter "x" of type "ConvertibleToInt" in function "__new__" (reportArgumentType)
```

I didn't find a good option to add type hinting in pandas DataFrames or Series.
Any suggestions?

**Comment by erictraut** [comment]
The pandas stubs are located [here](https://github.com/pandas-dev/pandas). If you have questions about these stubs, I recommend posting to that forum. They'll be able to help you with pandas-specific questions related to typing.

**Comment by GhostlyCipher** [comment, marked as answer]
Installing https://github.com/pandas-dev/pandas-stubs fixed this issue.
